src/librustc_mir/dataflow/move_paths/mod.rs

```diff
@@ -128,6 +128,43 @@ pub struct MoveData<'tcx> {
     pub rev_lookup: MovePathLookup<'tcx>,
 }
 
+impl<'tcx> MoveData<'tcx> {
+    // FIXME: This causes a strange error
```

Comment:
Of course it does, you're calling `all_paths_from::<&mut F>`, which further calls `all_paths_from::<&mut &mut F>` and so on. Please remove the commented code.

Reply:
This is also a version of `on_all_children_bits` that does not ignore types within destructors. I really need to straighten the story there.

Reply:
I removed this.